ported/omb/rdm_latency.c

```diff
@@ -0,0 +1,422 @@
+	if (ret != numprocs) {
+		FT_PRINTERR("fi_av_insert", ret);
+		return ret;
+	}
+
+	free(addr);
+
+	return 0;
+}
+
+int main(int argc, char *argv[])
+{
+	int i, peer;
+	int size, align_size;
+	char *s_buf, *r_buf;
+	uint64_t t_start = 0, t_end = 0;
+	int op, ret;
+	ssize_t fi_rc;
+
+	FT_Init(&argc, &argv);
+	FT_Rank(&myid);
+	FT_Job_size(&numprocs);
+
+	hints = fi_allocinfo();
+	if (!hints)
+		return -1;
+
+	while ((op = getopt(argc, argv, "h" INFO_OPTS)) != -1) {
+		switch (op) {
+		default:
+			ft_parseinfo(op, optarg, hints);
+			break;
+		case '?':
+		case 'h':
+			print_usage();
+			return EXIT_FAILURE;
+		}
+	}
+
+	hints->ep_attr->type	= FI_EP_RDM;
+	hints->caps		= FI_MSG | FI_DIRECTED_RECV;
+	hints->mode		= FI_CONTEXT | FI_LOCAL_MR;
+
+	if (numprocs != 2) {
+		if (myid == 0) {
+			fprintf(stderr, "This test requires exactly two processes\n");
+		}
+		FT_Finalize();
+		return -1;
+	}
+
+	/* Fabric initialization */
+	ret = init_fabric();
+	if (ret) {
+		fprintf(stderr, "Problem in fabric initialization\n");
+		return ret;
+	}
+
+	ret = init_av();
+	if (ret) {
+		fprintf(stderr, "Problem in AV initialization\n");
+		return ret;
+	}
+
+	/* Data initialization */
+	align_size = getpagesize();
+	assert(align_size <= MAX_ALIGNMENT);
+
+	s_buf = (char *) (((unsigned long) s_buf_original + (align_size - 1)) /
+				align_size * align_size);
+
+	r_buf = (char *) (((unsigned long) r_buf_original + (align_size - 1)) /
+				align_size * align_size);
+
+	if (myid == 0) {
+		fprintf(stdout, HEADER);
+		fprintf(stdout, "%-*s%*s\n", 10, "# Size", FIELD_WIDTH, "Latency (us)");
+		fflush(stdout);
+	}
+
+	for (size = 0; size <= MAX_MSG_SIZE; size = (size ? size * 2 : 1)) {
+		for (i = 0; i < size; i++) {
+			s_buf[i] = 'a';
+			r_buf[i] = 'b';
+		}
+
+		if (size > large_message_size) {
+			loop = loop_large;
+			skip = skip_large;
+		}
+
+		FT_Barrier();
+
+		if (myid == 0) {
+			peer = 1;
+			for (i = 0; i < loop + skip; i++) {
+				if (i == skip)
+					t_start = get_time_usec();
+
+				fi_rc = fi_send(ep, s_buf, size, NULL,
+						fi_addrs[peer], NULL);
+				assert(!fi_rc);
+				wait_for_data_completion(scq, 1);
+
+				fi_rc = fi_recv(ep, r_buf, size, NULL,
+						fi_addrs[peer], NULL);
+				assert(!fi_rc);
+				wait_for_data_completion(rcq, 1);
+			}
+
+			t_end = get_time_usec();
+		} else if (myid == 1) {
+			peer = 0;
+			for (i = 0; i < loop + skip; i++) {
+				fi_rc = fi_recv(ep, r_buf, size, NULL,
+						fi_addrs[peer], NULL);
+				assert(!fi_rc);
+				wait_for_data_completion(rcq, 1);
+
+				fi_rc = fi_send(ep, s_buf, size, NULL,
+						fi_addrs[peer], NULL);
+				assert(!fi_rc);
+				wait_for_data_completion(scq, 1);
+			}
+		}
+
+		if (myid == 0) {
+			double latency = (t_end - t_start) / (2.0 * loop);
+
+			fprintf(stdout, "%-*d%*.*f\n", 10, size, FIELD_WIDTH,
+					FLOAT_PRECISION, latency);
+			fflush(stdout);
+		}
+	}
+
+	FT_Barrier();
+
+	free_ep_res();
+
+	fi_close(&ep->fid);
```

Comment:
We should close EP before closing AV (Same comment for other benchmarks as well).